src/jquery.mask.js

```diff
@@ -472,11 +474,11 @@
 
     $.jMaskGlobals = $.jMaskGlobals || {};
     globals = $.jMaskGlobals = $.extend(true, {}, globals, $.jMaskGlobals);
-    
+
     // looking for inputs with data-mask attribute
     if (globals.dataMask) { $.applyDataMask(); }
 
-    setInterval(function(){
-        if ($.jMaskGlobals.watchDataMask) { $.applyDataMask(); }
-    }, globals.watchInterval);
+    if ($.jMaskGlobals.watchDataMask) {
```

Comment:
The ```if ($.jMaskGlobals.watchDataMask)``` must be inside of the setInterval in chase the value changes pragmatically so the settings must be re-evaluated every time.